src/createSpeechRecognition.ts

```diff
@@ -69,7 +75,13 @@ export const createSpeechlySpeechRecognition = (appId: string): SpeechRecognitio
 
     private readonly initialise = async (): Promise<void> => {
       if (!this.clientInitialised) {
-        await this.client.initialize()
+        const microphone = new BrowserMicrophone()
+        await microphone.initialize()
+        const { mediaStream } = microphone
+        if (mediaStream === null || mediaStream === undefined) {
+          throw ErrDeviceNotSupported
```

Comment:
In theory, we shouldn't reach this line as `microphone.initialize` should throw in these cases. This is just a precautionary step to (a) avoid silent failure and (b) reassure TypeScript that we aren't passing null to `attach`.